pkg/skaffold/docker/parse.go

```diff
@@ -108,6 +109,47 @@ func GetDockerfileDependencies(workspace string, r io.Reader) ([]string, error)
 	return filteredDeps, nil
 }
 
+func PortsFromDockerfile(r io.Reader) ([]string, error) {
+	res, err := parser.Parse(r)
+	if err != nil {
+		return nil, errors.Wrap(err, "parsing dockerfile")
+	}
+
+	// Check the dockerfile and the base.
+	ports := []string{}
+	for _, value := range res.AST.Children {
+		switch value.Value {
+		case from:
+			base := value.Next.Value
+			if base == "SCRATCH" {
```

Comment:
why couldn't you expose ports in a scratch image?

Reply:
you don't have ca-certs but internal services, etc.

Reply:
It's about the FROM check - we check your Dockerfile and the base image for ports. a SCRATCH base image won't have any PORTS exposed itself so we don't need to try to check it. We still check your Dockerfile for EXPOSES though.